tests/test_html_to_json.py

```diff
@@ -53,6 +69,8 @@ def test_html_in_json(file_path):
     # json dumps to escape html content
     assert any(json.dumps(HTML_FRAGMENT1) in s for s in content_in_json)
     assert any(json.dumps(HTML_FRAGMENT2) in s for s in content_in_json)
+    assert any(json.dumps(HTML_VARIANT1) in s for s in content_in_json)
+    assert any(json.dumps(HTML_VARIANT2) in s for s in content_in_json)
```

Comment:
I didn't notice this in the past reviews, but why not do all of these assertions in a more deterministic manner where you specifically check for the expected content in the expected JSON. Currently it seems to me that `content_in_json` ends up with the aggregate of all data, which means the test could pass even if the right content somehow ends up in the wrong files or incorrect structure. I don't think it would add much (if any) complexity to these tests to instead do something like:

```python
    assert get_json_content(file_path, 'fragment', 'main') == HTML_FRAGMENT1
    assert get_json_content(file_path, 'fragment2', 'main') == HTML_FRAGMENT2
    assert get_json_content(file_path, 'variantdir', 'variant1') == HTML_VARIANT1
    assert get_json_content(file_path, 'variantdir', 'variant2') == HTML_VARIANT2
```

where `get_json_content` helper is:

```python
def get_json_content(file_path, content_id, variant):
    json_dir = file_path / 'json_directory'
    content = json.load(open(str(json_dir / f'{content_id}.json')))['content']
    for entry in content:
        if entry['variant'] == variant:
            return entry['html']
```

Note: Doing the above probably removes the need for the `test_json_content_is_valid` test case since.